rasa/shared/core/training_data/structures.py

```diff
@@ -149,7 +149,12 @@ def _or_string(story_step_element: List[Event], e2e: bool) -> Text:
                 )
 
         result = " OR ".join(
-            [element.as_story_string(e2e) for element in story_step_element]
+            # FIXME: this breaks below because not
+            # all `as_story_string()` take a `e2e` argument.
```

Comment:
Can we change `_or_string(story_step_element: List[Event]` to `_or_string(story_step_element: List[UserUttered]`?

Reply:
I guess I didn't see that we explicitly check above that every element in the list if of type `UserUttered`. It's a `mypy` issue, because it doesn't infer the type correctly after the type guard. I've adapted using ab748f6307a